src/components/search-result-list/search-result-list-item/search-result-list-item.tsx

```diff
@@ -142,10 +143,10 @@ const SearchResultListItem: React.FC<SearchResultListItemProps> = ({
           <Link href={materialFullUrl}>{fullTitle}</Link>
         </h2>
 
-        {author && (
+        {author && item && (
           <p className="text-small-caption" data-cy="search-result-item-author">
             {`${t("byAuthorText")} ${author}`}
-            {workYear && ` (${workYear.year})`}
+            {getReleaseYear(item) ? ` (${getReleaseYear(item)})` : ""}
```

Comment:
Please consolidate these lines with
[this](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/341/files#diff-a09d47a1eb335411b7f4c67799d5a4d59bac0b55e8dcf3d3142c99f86d309378R348), [this](https://github.com/reload/dpl-react/blob/3a23ad1240698df192979453ce2e2111268b0915/src/components/reservation/helper.ts#L128) and [this](https://github.com/reload/dpl-react/blob/3a23ad1240698df192979453ce2e2111268b0915/src/core/utils/helpers/general.ts#L18)

Reply:
I incorporated `getManifestationPublicationYear()` helper into my function and renamed it + commented on it so that people know it's a function specific to search result page, as there we are very particular what years we want to show for what works.